### PR TITLE
[Type Hints] `nn.MetaLayer`

### PR DESCRIPTION
**Pull request by camillepradel**

### CODE REVIEW

File: torch_geometric/nn/meta.py [hunk 1]
```diff
@@ -104,9 +106,10 @@ def reset_parameters(self):
                 item.reset_parameters()
 
     def forward(
-            self, x: Tensor, edge_index: Tensor,
-            edge_attr: Optional[Tensor] = None, u: Optional[Tensor] = None,
-            batch: Optional[Tensor] = None) -> Tuple[Tensor, Tensor, Tensor]:
+        self, x: Tensor, edge_index: Tensor,
+        edge_attr: Optional[Tensor] = None, u: Optional[Tensor] = None,
+        batch: Optional[Tensor] = None
+    ) -> Tuple[Optional[Tensor], Optional[Tensor], Optional[Tensor]]:
```
Comment (rusty1s) [inline torch_geometric/nn/meta.py:112]:
```suggestion
    ) -> Tuple[Tensor, Optional[Tensor], Optional[Tensor]]:
```

File: torch_geometric/nn/meta.py [hunk 2]
```diff
@@ -104,9 +106,10 @@ def reset_parameters(self):
                 item.reset_parameters()
 
     def forward(
-            self, x: Tensor, edge_index: Tensor,
-            edge_attr: Optional[Tensor] = None, u: Optional[Tensor] = None,
-            batch: Optional[Tensor] = None) -> Tuple[Tensor, Tensor, Tensor]:
+        self, x: Tensor, edge_index: Tensor,
+        edge_attr: Optional[Tensor] = None, u: Optional[Tensor] = None,
+        batch: Optional[Tensor] = None
```
Comment (rusty1s) [inline torch_geometric/nn/meta.py:111]:
```suggestion
        batch: Optional[Tensor] = None, 
```

File: torch_geometric/nn/meta.py [hunk 3]
```diff
@@ -90,7 +90,9 @@ def forward(self, x, edge_index, edge_attr, u, batch):
         op = MetaLayer(EdgeModel(), NodeModel(), GlobalModel())
         x, edge_attr, u = op(x, edge_index, edge_attr, u, batch)
     """
-    def __init__(self, edge_model=None, node_model=None, global_model=None):
+    def __init__(self, edge_model: Optional[torch.nn.Module] = None,
+                 node_model: Optional[torch.nn.Module] = None,
+                 global_model: Optional[torch.nn.Module] = None):
```
Comment (rusty1s) [inline torch_geometric/nn/meta.py:95]:
```suggestion
                 global_model: Optional[torch.nn.Module] = None, ):
```

File: torch_geometric/nn/meta.py (hunk 1 above)
Comment (camillepradel) [reply torch_geometric/nn/meta.py]: Hi! I agree it makes more sense to remove the `Optional[]`. However, current `test_meta_layer` test fails with this type hinting, as we use a dummy model for `node_model` which always return `None` (and thus returned `x` value is `None` in that case).
Maybe I could update the dummy model to return `edge_attr` for `edge_model`, `x` for `node_model` and `u` for `global_model` ?

Comment (rusty1s) [reply torch_geometric/nn/meta.py]: Yeah, it probably makes sense to define them separately with the correct type hints.

**Review by rusty1s — APPROVED** [review]
Thank you so much!

**Pull request merged**